modules/core/shared/src/main/scala/gem/Asterism.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+import cats.Eq
+import cats.data.NonEmptyList
+
+import gem.enum.{ AsterismType, Instrument }
+
+import monocle.Prism
+import monocle.macros.{ Lenses, GenPrism }
+
+
+sealed trait Asterism extends Product with Serializable {
+
+  def asterismType: AsterismType
```

Comment:
Is the `AsterismType` used outside of serialization code? In general it would be nice to avoid propagating the discriminant into the model. I wonder if it would make sense to indicate these in the Schema somehow, to differentiate them from other enums.